compiler/lib/effects.ml

```diff
@@ -220,63 +344,96 @@ let cps_block ~st ~k pc block =
     match Addr.Map.find pc st.jc.closures_of_alloc_site with
     | to_allocate ->
         List.map to_allocate ~f:(fun (cname, jump_pc) ->
-            let jump_block = Addr.Map.find jump_pc st.blocks in
-            let fresh_params = List.map jump_block.params ~f:(fun _ -> Var.fresh ()) in
-            Let (cname, Closure (fresh_params, (jump_pc, fresh_params))))
+            let params =
+              let jump_block = Addr.Map.find jump_pc st.blocks in
+              if List.is_empty jump_block.params && Hashtbl.mem st.is_continuation jump_pc
```

Comment:
Can we have a small comment explaining what's going on here ?

Reply:
I have added a short comment above and updated the comment below